invenio_app_ils/circulation/api.py

```diff
@@ -211,6 +213,44 @@ def checkout_loan(
     return pid, loan
 
 
+def update_dates_loan(record, start_date=None, end_date=None, request_start_date=None, request_expire_date=None):
+    """Updates the dates of a loan."""
+    state = record["state"]
+    is_active_or_completed = state in CIRCULATION_STATES_LOAN_ACTIVE \
+        or state in CIRCULATION_STATES_LOAN_COMPLETED
+
+    data = copy(record)
+
+    if is_active_or_completed:
+        today = date.today().strftime("%Y-%m-%d")
+        if request_start_date or request_expire_date:
+            raise IlsException(description="Cannot modify request dates of an active or completed loan.")
+        if start_date:
+            if start_date > today:
+                raise InvalidParameterError(description="Start date cannot be in the future for active loans.")
+            data["start_date"] = start_date
+        if end_date:
+            data["end_date"] = end_date
+        if data["end_date"] < data["start_date"]:
```

Comment:
I would check this earlier before line 228, using the input params

Reply:
Both fields are optional, we have to make sure that the final state is correct. I can hardly see how this could be simplified without modifying the function's behavior.